include/aspect/newton.h

```diff
@@ -122,26 +123,53 @@ namespace aspect
       void parse_parameters (ParameterHandler &prm);
 
       /**
-       * A scaling factor for those terms of the Newton matrix that
-       * result from the linearization of the viscosity.
+       * A scaling factor used for scaling the
+       * derivative part of the Newton Stokes solver in the assembly.
        *
-       * See the get_newton_derivative_scaling_factor() function for an
-       * explanation of the purpose of this factor.
+       * The exact Newton matrix consists of the Stokes matrix plus a term
+       * that results from the linearization of the material coefficients.
+       * The scaling factor multiplies these additional terms. In a full
+       * Newton method, it would be equal to one, but it can be chosen
+       * smaller in cases where the resulting linear system has undesirable
+       * properties.
```

Comment:
So the main reason I add this parameter it to use it together with the `use_newton_residual_scaling_method`, because it helps the global convergence of the iteration. Because far way from the solution, you use mostly Picard, and the closer you get the the solution, the more Newton you use. And when it diverges for some reason it gets closer to Picard. It doesn't make it completely globally convergence, but it may help quite a bit. Although maybe part of this explanation should be at the `use_newton_residual_scaling_method` parameter. 

I can also update the documentation in a later pull request if you want.

Reply:
OK, I'll gladly let you make this update at a later time -- this is orthogonal to the current patch anyway.

Reply:
sure